src/dfmt/formatter.d

```diff
@@ -2148,13 +2148,18 @@ const pure @safe @nogc:
         return index < tokens.length && tokens[index].type == tokenType;
     }
 
-    bool onNextLine() @nogc nothrow pure @safe
+    bool onNextLine() @nogc pure @safe
     {
         import dfmt.editorconfig : OptionalBoolean;
+        import std.algorithm.searching : count;
+
+        if (config.dfmt_keep_line_breaks == OptionalBoolean.f || index <= 0)
+        {
+            return false;
+        }
+        const tokenEndLine = tokens[index - 1].line + tokens[index - 1].text.count('\n');
```

Comment:
keep this nothrow because it is used like a property everywhere, use `.text.representation.count('\n')` to avoid unicode exceptions